staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/defaulting/algorithm.go

```diff
@@ -21,8 +21,16 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// isNonNullalbeNull returns true if the item is nil AND it's nullable
+func isNonNullableNull(x interface{}, s *structuralschema.Structural) bool {
+	return x == nil && (s == nil || s.Generic.Nullable == false)
```

Comment:
If so, add a unit test that would have caught this?

Reply:
In that specific instance, it doesn't make a difference, because if we don't have a schema then we also don't have a default. But I agree it's confusing because it doesn't mean the same here. I'll change it.

Reply:
Changed, which also fixed the panic because now we know that we have a schema when isNonNullableNull is true. Test added.